src/Chainweb/Utils.hs

```diff
@@ -1074,9 +1076,22 @@ concurrentWith alloc inner params = do
 -- -------------------------------------------------------------------------- --
 -- Strict Tuple
 
+-- | First projections for strict tuples
+--
 sfst :: T2 a b -> a
 sfst (T2 a _) = a
 
+-- | Second projections for strict tuples
+--
 ssnd :: T2 a b -> b
 ssnd (T2 _ b) = b
 
+-- | Currying for functions of strict tuples
+--
+scurry :: forall a b c. ((T2 a b) -> c) -> a -> b -> c
+scurry k a b = k (T2 a b)
+
+-- | Uncurrying for functions of strict tuples
+--
+suncurry :: forall a b c. (a -> b -> c) -> T2 a b -> c
+suncurry k (T2 a b) = k a b
```

Comment:
👍

Reply:
Should this (as well as `scurry`, `ssnd`, and `sfst`) be inlined, or do we rely on GHC doing it anyways?

Reply:
I'll do it